docker/build_scripts/build-openssl.sh

```diff
@@ -16,7 +16,7 @@ check_var ${OPENSSL_HASH}
 check_var ${OPENSSL_DOWNLOAD_URL}
 
 OPENSSL_VERSION=${OPENSSL_ROOT#*-}
-OPENSSL_MIN_VERSION=1.1.1
+OPENSSL_MIN_VERSION=3.0.0
```

Comment:
Why change the minimum OpenSSL version ? Should still be 1.1.1

Reply:
should we still keep the minimum version same as old? and keep support through 1.x-3.x series?

Reply:
I understand that there will have to be two versions of openssl, the legacy and the current one. And in the case of the minimum version, it must be the one requested for each case.

Reply:
No, the minimum version, even for python 3.12 is still 1.1.1
This requirement started with python 3.10 and to simplify the workflow, it was set the same for all pythons versions.
So this shall remain 1.1.1 even for 3.8+

Reply:
In the recent version of the PR, I reverted the value.